File: Tp1/Is/Is.java
import java.util.Scanner;

public class Is{
	//Metodo de teste de vogais
	public static boolean isVogais(String line){
		boolean result = false;
		for(int i = 0; i < line.length(); i++){
			if(line.charAt(i) == 'a' || line.charAt(i) == 'e' || line.charAt(i) == 'i' || line.charAt(i) == 'o' ||
			   line.charAt(i) == 'u' || line.charAt(i) == 'A' || line.charAt(i) == 'E' || line.charAt(i) == 'I' || 
			   line.charAt(i) == 'O' || line.charAt(i) == 'U'){
				result = true;
			} else {
				result = false;
				break;
			}
		}
		return result;
	}
	//Metodo de teste de consoantes
	public static boolean isConsoantes(String line){
		boolean result = false;
		String lineLower = line.toLowerCase();
		for(int i = 0; i < line.length(); i++){
			if(lineLower.charAt(i) == 'b' || lineLower.charAt(i) == 'c' || lineLower.charAt(i) == 'd' || lineLower.charAt(i) == 'f' ||
			   lineLower.charAt(i) == 'g' || lineLower.charAt(i) == 'j' || lineLower.charAt(i) == 'k' || lineLower.charAt(i) == 'l' ||
			   lineLower.charAt(i) == 'm' || lineLower.charAt(i) == 'n' || lineLower.charAt(i) == 'n' || lineLower.charAt(i) == 'p' ||
			   lineLower.charAt(i) == 'q' || lineLower.charAt(i) == 'r' || lineLower.charAt(i) == 's' || lineLower.charAt(i) == 't' ||
			   lineLower.charAt(i) == 'v' || lineLower.charAt(i) == 'w' || lineLower.charAt(i) == 'x' || lineLower.charAt(i) == 'z' ){
				   result = true;
			   } else {
				   result = false;
				   break;
			   }
		}
		return result;
	}
	//Metodo de teste sem e um numero
	public static boolean isNum(String line){
		boolean result = false;
		for(int i = 0; i < line.length(); i++){
			if(line.charAt(i) >= '1' && line.charAt(i) <= '9'){
				result = true;
			} else {
				result = false;
				break;
			}
		}
		return result;
	}
	//Metodo de teste se e um numero real
	public static boolean isReal(String line){
		boolean result = false;
		int cont = 0;
		for(int i = 0; i < line.length(); i++){
			if(line.charAt(i) >= '1' && line.charAt(i) <= '9'){
				result = true;
			} else if((line.charAt(i) == ',' || line.charAt(i) == '.') && cont < 1){
				result = true;
				cont++;
			} else {
				result = false;
				break;
			}
		}
		return result;
	}

	public static boolean isFim(String frase){
		return (frase.length() == 3 && frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M');
	}

	public static void main(String args[]){
		Scanner entrada = new Scanner(System.in);
		
		String line = entrada.nextLine();
		do{
			//Chamada de metodo de teste de vogais
			if(isVogais(line)){
				System.out.print("SIM ");
			} else {
				System.out.print("NAO ");
			}
			
			//Chamada de metodo de teste de consoantes
			if(isConsoantes(line)){
				System.out.print("SIM ");
			} else {
				System.out.print("NAO ");
			}
			
			//Chamada de metodo de teste se e um numero
			if(isNum(line)){
				System.out.print("SIM ");
			} else {
				System.out.print("NAO ");
			}
			//Chamada de metodo de teste numero Real
			if(isReal(line)){
				System.out.println("SIM ");
			} else {
				System.out.println("NAO ");
			}

			line = entrada.nextLine();
		}while(!isFim(line));	
	}
}

File: Tp1/Is/Is.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stdlib.h>
#include <ctype.h>

bool isVogais(char frase[1000]){
		bool result = false;
		
		for(int i = 0; i < strlen(frase) - 1; i++){
			if(frase[i] == 'a' || frase[i] == 'e' || frase[i] == 'i' || frase[i] == 'o' ||
			   frase[i] == 'u' || frase[i] == 'A' || frase[i] == 'E' || frase[i] == 'I' || 
			   frase[i] == 'O' || frase[i] == 'U'){
				result = true;
			} else {
				result = false;
				break;
			}
		}
		return result;
}

bool isConsoantes(char frase[1000]){
		bool result = false;
		char fraseLower[1000];
		int i = 0;
		while(frase[i] != '\0'){
			fraseLower[i] = tolower(frase[i]);
			i++;
		}
		for(i = 0; i < strlen(frase) - 1; i++){
			if(frase[i] == 'b' || frase[i] == 'c' || frase[i] == 'd' || frase[i] == 'f' ||
			   frase[i] == 'g' || frase[i] == 'j' || frase[i] == 'k' || frase[i] == 'l' ||
			   frase[i] == 'm' || frase[i] == 'n' || frase[i] == 'n' || frase[i] == 'p' ||
			   frase[i] == 'q' || frase[i] == 'r' || frase[i] == 's' || frase[i] == 't' ||
			   frase[i] == 'v' || frase[i] == 'w' || frase[i] == 'x' || frase[i] == 'z' ){
				   result = true;
			   } else {
				   result = false;
				   break;
			   }
		}
		return result;
}

bool isNum(char frase[1000]){
		bool result = false;
		int i = 0;
		for(int i = 0; i < strlen(frase) - 1; i++){
			if(frase[i] >= '1' && frase[i] <= '9'){
				result = true;
			} else {
				result = false;
				break;
			}
		}
		return result;
}

bool isReal(char frase[1000]){
		bool result = false;
		int cont, i = 0;
		for(i = 0; i < strlen(frase) - 1; i++){
			if(frase[i] >= '1' && frase[i] <= '9'){
				result = true;
			} else if((frase[i] == ',' || frase[i] == '.') && cont < 1){
				result = true;
				cont++;
			} else {
				result = false;
				break;
			}
		}
		return result;
}

bool isFim(char frase[100]){
	return (frase[0] == 'F' && frase[1] == 'I' && frase[2] == 'M');
}

int main(void){
	char frase[1000];
	fgets(frase, 1000, stdin);
	fflush(stdin);
	while(!isFim(frase)){
			//Chamada de metodo de teste de vogais
			if(isVogais(frase)){
				printf("SIM ");
			} else {
				printf("NAO ");
			}
			
			//Chamada de metodo de teste de consoantes
			if(isConsoantes(frase)){
				printf("SIM ");
			} else {
				printf("NAO ");
			}
			
			//Chamada de metodo de teste se e um numero
			if(isNum(frase)){
				printf("SIM ");
			} else {
				printf("NAO ");
			}
			//Chamada de metodo de teste numero Real
			if(isReal(frase)){
				printf("SIM\n");
			} else {
				printf("NAO\n");
			}
			fgets(frase, 1000, stdin);
			fflush(stdin);
	}
	return 0;
}
